types/wlr_data_device.c

```diff
@@ -309,25 +285,30 @@ static void seat_client_selection_data_source_destroy(
 		}
 	}
 
-	seat->selection_source = NULL;
+	seat->selection_data_source = NULL;
 
 	wl_signal_emit(&seat->events.selection, seat);
 }
 
 void wlr_seat_set_selection(struct wlr_seat *seat,
 		struct wlr_data_source *source, uint32_t serial) {
-	if (seat->selection_source &&
+	if (source) {
+		assert(source->send);
+		assert(source->cancel);
```

Comment:
We need to be careful about what is required of the data source and fail as fast as possible when it is not present.

Reply:
Yeah, that's the intent of these checks.